File: test/features/page_objects/signInPage.js
// Page class for SignIn page
'use strict'

class SignInPage {

constructor () {
    this.heading = element(by.css('.heading-xlarge'));
    this.instructions = element(by.css('p.lede'));
    this.usernameField = element(by.css('#username'));
    this.passwordField = element(by.css('#password'));
    this.signIn = element(by.css('input[value="Sign in"]'));
    this.phase_banner = element(by.css('phase-banner'));
}

    enterCredentials(username,password) {
        this.usernameField.sendKeys(username)
        this.passwordField.sendKeys(password)
    }


    login(username,password) {
       this.enterCredentials(username,password)
        this.signIn.click();
    }
};

module.exports = new SignInPage();

File: admin/test/features/support/page_objects/sign_in_page.rb
class SignInPage < SitePrism::Page
  set_url '/sign-in'

  element :heading, '.heading-xlarge', text: 'Check Development Service'
  element :instructions, 'p.lede', text: 'Sign-in to access the Check Development Service'
  element :username_field, '#username'
  element :password_field, '#password'
  element :sign_in, 'input[value="Sign in"]'
  element :contact, "a[href='/contact']"
  section :phase_banner, PhaseBanner, '.phase-banner'


  def enter_credentials(username, password)
    username_field.set username
    password_field.set password
  end

  def login(username,password)
    enter_credentials(username,password)
    sign_in.click
  end

end
